advice_for_future_giraf/process_advice.md

```diff
@@ -0,0 +1,38 @@
+
+* Every group gets to see more of the code because they implement full features.
+* Less pipelining is needed as features are not made in parts by different groups.
+* People are motivated to finish their tasks because they can see their implementations immediately in the application.
+
+After having used the full stack approach for this year's Giraf project we believe that it is a good fit for this kind of semester project. We would therefore urge future groups to keep on using this approach and not go back to the specialist approach.<p>
+This approach does have downsides that need to be managed still. However, they are not really intrinsic to the approach itself.<br>
+Be careful to consider the following points when you think out the process for your own project:
+
+* How do you find dependencies between user stories? 
+    + How do you handle dependencies once discovered?
+* How do you make sure that one group's work is not made obsolete by another's? (Especially when the two groups are working in the same files.)
+
+We found that handling dependencies needs constant work with our process. It is quickly forgotten by individual groups that their work might affect others or that others might depend on one user story to implement their own if people are not made aware of this constantly.
+
+## Do Mention Everything Explicitly
+Throughout the 2019 Giraf project we found that the only way to make sure that every group followed the process correctly was to tell them every part of it (sometimes repeatedly). While this might sound trivial, there are a lot of small things that can be seen as implicit to some people and if the process group finds something to be self-evident they might not mention it. However small any part of the process might be, it should always be mentioned explicitly to everyone in the project.<br>
+If you decide to use some form of frequent meetings (e.g. daily Scrum of Scrums stand up meetings) this is a good place to remind people of small things related to process. Even if it might be redundant at the end, at least you know that people are on the same page.
+
+## Do Plan All Meetings of a Sprint at the Beginning of it
+It is important that everyone on the Giraf project knows when certain milestones and meetings occur during a sprint. It is all too easy to schedule these (especially frequent recurring meetings such as Scrum of Scrums stand up meetings) on a week by week basis. However, we found that with this way of scheduling, some meetings were scheduled too late for people to notice. This caused some stress for the different groups as they could not schedule their work around meetings far in advance. Later in the project we began scheduling all meetings of a sprint at the start of the sprint itself. This meant that everyone knew when all meetings were happening and also meant that no meetings had to be delayed or skipped because of scheduling them too late.
```

Comment:
```suggestion
It is important that everyone on the Giraf project knows when certain milestones and meetings occur during a sprint. It is all too easy to schedule these (especially frequent recurring meetings such as Scrum of Scrums stand up meetings) on a week by week basis. However, we found that with this way of scheduling, some meetings were scheduled too late for people to notice. This caused some stress for the different groups as they could not schedule their work around meetings far in advance. Later in the project we began scheduling all meetings of a sprint at the beginning of the sprint itself. This meant that everyone knew when all meetings were happening and it also meant that no meetings had to be delayed or skipped because of annnouncing them too late.
```